tests/unit/Controller/SocialApiControllerTest.php

```diff
@@ -73,99 +75,31 @@ public function setUp() {
 			$this->config,
 			// $this->initialStateService,
 			$this->languageFactory
-
 		);
-	}
-
-	public function testGetFacebookAvatarId() {
-		$this->service->expects($this->once())
-			->method(getSocialConnector)
-			->with($this->equalTo(array('facebook','4')),
-				$this->equalTo('avatar'))
-			->will($this->returnValue($response));
-
-		$result = $this->controller->getSocialConnector(array('facebook','4'),'avatar');
 
-		// social connector is not null
-		$this->assert(($result) > 4);
-		
-		// social connector starts with http
-		$this->assert(substr_compare($result, 'http', 0, 4));
-	}
-
-	public function testGetFacebookAvatarFaultyId() {
-		$this->service->expects($this->once())
-			->method(getSocialConnector)
-			->with($this->equalTo(array('facebook','alphan0meric')),
-				$this->equalTo('avatar'))
-			->will($this->returnValue($response));
+		$this->entry       = $this->createMock(IEntry::class);
+		$this->addressbook = $this->createMock(IAddressBook::class); // FIXME: can I have a dummy address book or do I need integration tests for that?
```

Comment:
I am starting to look into phpunit - can I have a dummy address book or do I need integration tests for that?

Reply:
You should be able to mock an addressbook or at least create a dummy one :)

Btw, we should maybe finish the Controller before diving into unit testing.
There are lots of logic that should be cleaned up ;)

Reply:
But it might be lost in the previous comments, do we want to create a new pull request on a nextcloud branch so it's cleaner ?

Reply:
Okay, did a big review pass :)
After that, what do you think about a new pr? We discussed things a lot here, so the issue is quite polluted :see_no_evil:

Reply:
Thank you for all your comments! I will try to follow things here, but if you would prefer a new pull request that's also fine for me.

Reply:
It's all up to you! Because I left lots of comments, github tends to have a cluttered interface in those cases. So if you feel overwhelmed by those and feel the ticket is a bit messy, feel free to create a branch on the contacts repo and open the same pull request so we can close this one (you have the permissions now that you're part of Nextcloud's org :wink: )

Reply:
For me its perfectly fine here. I like having everything here at one place (and I feel there are going to be more comments needed when starting with the tests). If there are no other drawbacks, I have no problem with this long history.

Reply:
All good then!

Reply:
Changed the easy tasks, will continue the rest later (until everything is marked resolved here, I have it on my list).